### PR TITLE
Rocket launcher fixes, adds suicide_act

### LINKED ISSUE #41215: Syndicate Rocket Launcher and its rockets HATE tacticool reloads
Did a bit of testing at round ID 96257 in VR at Bagil. Then followed it up with some testing at 96259 on Bagil again (Thanks Imsxz), before switching over to the latest compile locally to confirm some of my documentations.

The steps to breaking the weapon (and its ammo) are simple; get a rocket launcher, now put a rocket in the already loaded rocket launcher (tactically reloading it), the sprite will disappear. Now unloading that rocket while using it in your hand (while its hopefully still there) will make the sprite re-appear.

Now the kicker is the fact that this will break both the rockets that were inserted and ejected during the reload. The one that was loaded will now cause all rocket launchers it is loaded into disappear, and the one that was ejected from the tac reload will change the RPG sprite into an unloaded one (with the red light). Put any other rocket that was untouched into the same rocket launcher and it'll perform accordingly, apart from the fact that it doesn't change the sprite from loaded to unloaded when actually unloaded. (Oh yeah, and tactically reloading more rockets into the invisible rocket launcher doesn't seem to do anything, at least from my testing. And firing it makes the sprite re-appear and functions normally with all the healthy rockets)

Also you can eject a rocket bullet out of a rocket, because apparently rockets themselves are considered ammoboxes, and the RPG is considered an automatic for whatever reason. _I really wish I understood how anything made sense here. Also I don't even know how coding works so don't ask me for anything, these are mere interesting findings I've dug up._

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
fix: You can no longer try to tactically reload rocket launchers, making it become invisible and breaking it's rocket.
fix: You can no longer unload a rocket with TK
tweak: Suiciding while holding a rocket launcher will now cause you to blow yourself up with it.
/:cl:

Fixes #41215

### CODE REVIEW

File: code/modules/projectiles/guns/ballistic/launchers.dm [hunk 1]
```diff
@@ -111,15 +127,31 @@
...
+		if(!QDELETED(user)) //if they weren't gibbed by the explosion, take care of them for good.
+			user.gib()
+		return MANUAL_SUICIDE
+	else
+		user.visible_message("<span class='warning'>[user] sticks [src] in [user.p_their()] mouth, ready to pull the trigger...</span>", \
+			"<span class='userdanger'>You stick [src] in your mouth, ready to pull the trigger...</span>")
+		sleep(5)
+		shoot_with_empty_chamber(user)
+		sleep(20)
+		user.visible_message("<span class='warning'>[user]'s looks about the room realizing [user.p_theyre()] still there. [user.p_they(TRUE)] proceed to shove [src] further down their throat and choke [user.p_them()]self with it!</span>", \
```
Comment (MrDoomBringer) [inline code/modules/projectiles/guns/ballistic/launchers.dm]: ( ͡° ͜ʖ ͡°)

Comment (evsey9) [reply code/modules/projectiles/guns/ballistic/launchers.dm]: is this fetish code

**Comment by epochayur** [comment]
\>not "[x] tries to rocket jump!"

**Comment by MrDoomBringer** [comment]
Rocket jump and launch the player up at mach 6

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**